pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -198,6 +200,15 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 		}
 	}
 
+	// re-apply host status from annotation
+
+	obj := r.getHostFromAnnotation(host)
+	// Manually restore data.
```

Comment:
this should be done only if you know that you do not have a status. It should be in line 209

Reply:
True. Changed it.